agent-core/src/main/java/com/newrelic/agent/android/HttpHeaders.java

```diff
@@ -0,0 +1,47 @@
+package com.newrelic.agent.android;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class HttpHeaders {
+
+    private static HttpHeaders instance = null;
+    private final Set<String> httpHeaders;
+    private static final String OPERATION_NAME = "X-APOLLO-OPERATION-NAME";
+    private static final String OPERATION_TYPE = "X-APOLLO-OPERATION-TYPE";
+
+    private static final String OPERATION_ID = "X-APOLLO-OPERATION-ID";
+
+
+    private HttpHeaders() {
+        httpHeaders = new HashSet<>();
+        httpHeaders.add(OPERATION_NAME);
+        httpHeaders.add(OPERATION_ID);
+        httpHeaders.add(OPERATION_TYPE);
+    }
+
+    public static HttpHeaders getInstance() {
+        if (instance == null) {
+            instance = new HttpHeaders();
+        }
+        return instance;
+    }
+
+    public void addHttpHeaderAsAttribute(String httpHeader) {
+        httpHeaders.add(httpHeader);
+    }
+
+    public void removeHttpHeaderAsAttribute(String httpHeader) {
```

Comment:
I added the removeHttpHeaderAsAttribute method to give our customers the flexibility to remove specific HTTP headers at runtime for particular HTTP requests. This allows them to prevent the headers from being captured if they don't want to include them for specific purposes.